gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -516,4 +518,73 @@ def save_metrics_reports(metrics_group_list: List[GtsfmMetricsGroup], metrics_pa
         metrics_group_list, os.path.join(metrics_path, "gtsfm_metrics_report.html"), None
     )
 
-    
\ No newline at end of file
+def merge_two_partition_results(poses1: dict[int, Pose3], 
+                                poses2: dict[int, Pose3]) -> dict[int, Pose3]:
+    """
+    Merges poses from two partitions (dictionaries mapping camera index to Pose3).
+
+    Assumes poses1 are relative to frame 'a' and poses2 are relative to frame 'b'.
+    Finds the transformation 'aTb' (from frame 'b' to frame 'a') using
+    cameras present in both dictionaries (overlapping nodes). Then transforms
+    all poses from partition 2 into frame 'a' and merges them with partition 1.
+
+    Args:
+        poses1: Dictionary {camera_index: pose_in_frame_a}.
+        poses2: Dictionary {camera_index: pose_in_frame_b}.
+
+    Returns:
+        A merged dictionary {camera_index: pose_in_frame_a}.
+
+    Raises:
+        ValueError: If no overlapping cameras are found between the two partitions.
+    """
+
+    overlapping_keys = []
```

Comment:
No function should be larger than 4 lines (without comments)

Reply:
Done.